ui/widgets/checkboxradio.js

```diff
@@ -177,8 +177,8 @@ $.widget( "ui.checkboxradio", [ $.ui.formResetMixin, {
 
 		if ( this.options.icon && this.type === "checkbox" ) {
 
-			// We add ui-state-highlight to change the icon color
-			this._toggleClass( this.icon, null, "ui-icon-check ui-state-highlight", checked )
+			// We add ui-state-checked to change the icon color based on ui-state-highlight colors
```

Comment:
I'd drop the reference to `ui-state-highlight` since that's a ThemeRoller-specific thing.